### PR TITLE
Add allow_net_admin field to google_container_cluster resource

### PR DESCRIPTION
**Pull request by jeperetz**
Fixes issue https://github.com/hashicorp/terraform-provider-google/issues/15136

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `allow_net_admin` field to `google_container_cluster` resource
```

### CODE REVIEW

**Review by NickElliot — CHANGES_REQUESTED** [review]
At initial pass looks good. Could you also update the [documentation](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/website/docs/r/container_cluster.html.markdown) to include the new field? 

While that update is made I'm going to verify that the one-way required_with doesn't produce undesired forced recreations as it appears to be an unprecedented configuration. It may end up being better to remove that terraform side validation just to be sure, because as the allow_net_admin field is part of the Autopilot block, I assume that the API will reject a request that supplies only allow_net_admin, or it will simply do nothing if included on its own.

**Review by NickElliot — COMMENTED** [review]
LGTM! to @hao-nan-li for final approval

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -2750,6 +2769,29 @@ func resourceContainerClusterUpdate(d *schema.ResourceData, meta interface{}) er
 		log.Printf("[INFO] GKE cluster %s's cluster-wide autoscaling has been updated", d.Id())
 	}
 
+	if d.HasChange("allow_net_admin") {
```
Comment (hao-nan-li) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:2772]: Is this field updatable? If so can you add an update test?

Comment (jeperetz) [reply mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:2772]: Yes the field is updatable. I added update steps to the test to toggle the feature off and on again.

**Review by hao-nan-li — APPROVED** [review]

**Review by NickElliot — APPROVED** [review]

**Pull request merged**